### PR TITLE
better fix for starve /sleep

### PR DESCRIPTION
**Pull request by 6fears7**

### CODE REVIEW

File: Story/RainMeadow.StoryHooks.cs [hunk 1]
```diff
@@ -329,16 +329,10 @@ private void RainWorldGame_GoToDeathScreen(On.RainWorldGame.orig_GoToDeathScreen
             {
                 foreach (OnlinePlayer player in OnlineManager.players)
                 {
-                    if (OnlineManager.lobby.isOwner)
-                    {
-                        RPCs.GoToDeathScreen();
```
Comment (Chrometurtle) [inline Story/RainMeadow.StoryHooks.cs:334]: This is an optimization

**Review by Chrometurtle — CHANGES_REQUESTED** [review]
You're removing an optimization

File: Story/RainMeadow.StoryHooks.cs (hunk 1 above)
Comment (6fears7) [reply Story/RainMeadow.StoryHooks.cs:334]:
```
                  if (player.id == OnlineManager.lobby.owner.id)
                    {
                        RPCs.GoToDeathScreen();
                    }
                    else
                    {

                        player.InvokeRPC(RPCs.GoToDeathScreen);
                    }
```

Keeps optimization and prevents unintended repeat invocations

Comment (Chrometurtle) [reply Story/RainMeadow.StoryHooks.cs:334]: oh? What was causing the repeat invocations. There should have only ever been 1 lobby owner

Comment (6fears7) [reply Story/RainMeadow.StoryHooks.cs:334]: I believe it was from having a separate function called, "MovePlayersToDeathScreen" that simply iterated over all players and was called during this looping, so players were getting double-dosed with calls to move to sleep / death and dying of starvation rapidly. 

https://github.com/henpemaz/Rain-Meadow/pull/186/files

This solution should provide the same behavior while preventing repeats.

**Review by Chrometurtle — APPROVED** [review]

File: Story/RainMeadow.StoryHooks.cs (hunk 1 above)
Comment (Chrometurtle) [reply Story/RainMeadow.StoryHooks.cs:334]: yup checks out.

**Pull request merged**